sites/en/intro-to-rails/ruby_language.step

```diff
@@ -48,51 +48,78 @@ my_variable * 3
 
   step do
     console <<-RUBY
-fruits = fruits + ["orange"]
-fruits = fruits - ["kiwi"]
+      fruits = fruits + ["orange"]
+      fruits = fruits - ["kiwi"]
     RUBY
 
     message "`+` and `-` are called operators. We can use them with the array of fruits just like we can use them with numbers."
 
   end
 
+  step do
+    message "A **hash** is a dictionary-like collection of unique **keys** and their **values**."
+
+    console <<-RUBY
+      fruit_dictionary = { strawberry: "a sweet soft red...", plum: "an oval, purple fruit...", orange: "a round juicy citrus fruit..." }
+    RUBY
+
+    message "We're creating a hash which associates the name of a fruit (**key**) with the definition of the fruit (**value**)."
+  end
+
+  step do
+    console <<-RUBY
+      fruit_dictionary[:strawberry]
+    RUBY
+
+    message "Here we're using the **key**, `:strawberry`, to look up the associated **value**, the definition of 'strawberry' in the fruit dictionary."
+  end
+
   step do
     message "Everything in Ruby has a **class**. Type this into IRB:"
 
     console_without_message <<-RUBY
-7.class
-"kiwi".class
-fruits.class
-RUBY
+      7.class
+      "kiwi".class
+      fruits.class
+    RUBY
 
-    message "These are the three data types introduced so far: **Fixnum** (numbers), **String** (text), and **Array** (lists)."
+    message "These are the four data types introduced so far: **Fixnum** (numbers), **String** (text), **Array** (lists), and **Hash** (dictionaries)."
   end
 
   step do
     message "Each class has different **methods** that can be used on **instances** of that class."
 
     console_without_message <<-RUBY
-fruits.length
-fruits.first
+      fruits.length
+      fruits.first
     RUBY
 
     message "You can see all the methods available for an object:"
     console_without_message <<-RUBY
-fruits.methods
+      fruits.methods
     RUBY
 
-    message "And **chain** methods together:"
+    message "And you can call multiple methods in a row:"
```

Comment:
Not a big deal but I think its helpful to have some mention of "chaining" here well assuming there isn't already on this page because "method chaining" is such a common term. I guess its obvious but figure its helpful to mention something like "also referred to as method chaining"?

Reply:
Yah, I thought about this a bit.  For me, it's a level of detail that for beginners, in particular, will likely be forgotten since there is so much information coming at them, which is why I decided to keep this change from the Boston docs.